spring-session/src/main/java/org/springframework/session/jdbc/JdbcOperationsSessionRepository.java

```diff
@@ -0,0 +1,483 @@
+import org.springframework.context.ApplicationEvent;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.ApplicationEventPublisherAware;
+import org.springframework.core.serializer.Deserializer;
+import org.springframework.core.serializer.Serializer;
+import org.springframework.core.serializer.support.DeserializingConverter;
+import org.springframework.core.serializer.support.SerializingConverter;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.expression.Expression;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.springframework.jdbc.core.JdbcOperations;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.PreparedStatementSetter;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.lob.DefaultLobHandler;
+import org.springframework.jdbc.support.lob.LobHandler;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.session.ExpiringSession;
+import org.springframework.session.FindByIndexNameSessionRepository;
+import org.springframework.session.MapSession;
+import org.springframework.session.Session;
+import org.springframework.session.events.SessionCreatedEvent;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * A {@link org.springframework.session.SessionRepository} implementation that uses
+ * Spring's {@link JdbcOperations} to store sessions in a relational database. This
+ * implementation does not support publishing of
+ * {@link org.springframework.session.events.SessionDestroyedEvent} and its subclasses.
+ * <p>
+ * An example of how to create a new instance can be seen below:
+ *
+ * <pre class="code">
+ * JdbcTemplate jdbcTemplate = new JdbcTemplate();
+ *
+ * JdbcOperationsSessionRepository sessionRepository = new JdbcOperationsSessionRepository(jdbcTemplate);
+ * </pre>
+ *
+ * For additional information on how to create and configure a JdbcTemplate, refer to the
+ * <a href="http://docs.spring.io/spring/docs/current/spring-framework-reference/html/">
+ * Spring Framework Reference Documentation</a>.
+ * <p>
+ * By default, this implementation uses <code>SPR_SESSION</code> table to store sessions.
+ * SQL scripts to initialize the default schema are packaged as <code>schema-*.sql</code>,
+ * where <code>*</code> is the target database type. Note that the table prefix can be
+ * customized using the {@link #setTablePrefix(String)} method.
+ *
+ * @author Vedran Pavic
+ * @since 1.2.0
+ */
+public class JdbcOperationsSessionRepository
+		implements FindByIndexNameSessionRepository<JdbcOperationsSessionRepository.JdbcSession>, ApplicationEventPublisherAware {
+
+	private static final Log logger = LogFactory.getLog(JdbcOperationsSessionRepository.class);
+
+	private static final String DEFAULT_TABLE_PREFIX = "SPR_";
```

Comment:
I agree with @copa2 about [the table prefix](https://github.com/spring-projects/spring-session/pull/379#issuecomment-189323488).

If you need to allow customizing the queries, then allow the entire query to be injected.